admin/release.py

```diff
@@ -559,23 +565,28 @@ def upload_rpms(scratch_directory, target_bucket, version, build_server):
         target_distro_suffix = ""
 
     operating_systems = [
-        {'distro': 'fedora', 'version': '20', 'arch': 'x86_64'},
-        {'distro': 'centos', 'version': '7', 'arch': 'x86_64'},
-        {'distro': 'ubuntu', 'version': '14.04', 'arch': 'amd64'},
+        {'distro': 'fedora', 'version': '20', 'arch': 'native'},
+        {'distro': 'centos', 'version': '7', 'arch': 'native'},
+        {'distro': 'ubuntu', 'version': '14.04', 'arch': 'native'},
+        {'distro': 'ubuntu', 'version': '15.04', 'arch': 'native'},
     ]
 
     for operating_system in operating_systems:
+        for package_type, distribution_names in PACKAGE_TYPE_MAP.items():
+            if operating_system['distro'] in distribution_names:
+                architecture = ARCH[operating_system['arch']][package_type]
```

Comment:
This loop would probably make a good stand-alone function.  Probably in `admin/packaging.py` (using these magic strings, "native" and "all", as an API between modules is meh).  With error reporting, probably, instead of fall-off-the-end.